go-controller/hybrid-overlay/cmd/hybrid-overlay-node/hybrid-overlay-node.go

```diff
@@ -103,12 +104,7 @@ func runHybridOverlay(ctx *cli.Context) error {
 	}
 
 	f.Start(stopChan)
-	go func() {
-		err := n.Run(stopChan)
-		if err != nil {
-			klog.Error(err)
-		}
-	}()
+	n.Run(stopChan, &sync.WaitGroup{})
 
 	// run until cancelled
 	<-ctx.Context.Done()
```

Comment:
The `WaitGroup` shouldn't be anonymous, we should define it `wg := &sync.WaitGroup{}` and wait on it after the signal handling logic. `wg.Wait()`